main.go

```diff
@@ -180,6 +188,34 @@ func (c MainController) authMiddleware(h http.Handler) http.Handler {
 	})
 }
 
+func (c MainController) authAdminMiddleware(h http.Handler) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		user, authed := getAuthedADBUser(c.db, r)
+		if !authed {
+			// Delete the cookie if it doesn't auth.
+			c := &http.Cookie{
+				Name:     "auth-session",
+				Path:     "/",
+				MaxAge:   -1,
+				HttpOnly: true,
+			}
+			http.SetCookie(w, c)
+
+			http.Redirect(w, r, "/login", http.StatusFound)
+			return
+		}
+
+		if !user.Admin {
+			// Add 403 status
```

Comment:
Maybe it's good enough to flash that the user doesn't have permission to access that route.

Reply:
We probably don't want the user to be sitting on that page, or "be allowed to end up there", when they don't have permission. Ideally, we might want to send them to a new page that shows conveys that they don't have access to that page. Something like a "403" route.

The `apiAdminAuthMiddleware` actually only flashes that they don't have permission.

Reply:
Ah, I see. Want to create a 403 route?

Reply:
Yea. I can do that.